### ISSUE TITLE
Provider Config "auth_method_id" Yields Unhelpful Error Response On 404

### ISSUE THREAD

**Issue by DanielleMiu** [hashicorp/terraform-provider-boundary#276]
### Terraform Version
```
% tf version
Terraform v1.2.3
on darwin_arm64
+ provider registry.terraform.io/hashicorp/boundary v1.0.12
```

### Affected Resource(s)
Please list the resources as a list, for example:
- `provider "boundary"`

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
provider "boundary" {
	addr                            = "http://127.0.0.1:9200"
	auth_method_id                  = "ampw_1234567890"
	password_auth_method_login_name = "admin"
	password_auth_method_password   = "password"
}
```

### Expected Behavior
a better error response describing the error, rather than a raw api response
```
╷
│ Error: unknown auth_method_id, api returned 404
│ 
│   with provider["registry.terraform.io/hashicorp/boundary"],
│   on main.tf line 12, in provider "boundary":
│   12: auth_method_id                  = "ampw_1234567890"
│ 
╵
```
this error is just a suggestion

### Actual Behavior
outputs the 404 response directly from boundary
```
╷
│ Error: {"kind":"NotFound", "message":"Resource not found."}
│ 
│   with provider["registry.terraform.io/hashicorp/boundary"],
│   on main.tf line 10, in provider "boundary":
│   10: provider "boundary" {
│ 
╵
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. create (or modify) a `main.tf` to include (at least)
  - `required_providers`
  - `provider "boundary"`
  - at least one boundary resource (to force a connection)
2. run `terraform plan`

sample tf file to trigger the error: [main.txt](https://github.com/hashicorp/terraform-provider-boundary/files/9628712/main.txt)
(note: `.txt` used as github does not allow `.tf`)

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:
- none that I know of

**Comment by louisruch** [comment]
The specific error encountered here comes from h htttps://github.com/hashicorp/terraform-provider-boundary/blob/main/internal/provider/provider.go#L162
The tf provider at this point does not know the issue is the wrong auth methoid id, all it knows is Boundary returned a 404 during the authenticate call. But we should probably be better about wrapping errors with something like `op` as we do in Boundary core, as this will improve the traceability of the errors we get. In this example we could log the error occurred during auth for example.

**Comment by ddebko** [comment]
Fixed in v1.1.2
